wrappers/services.go

```diff
@@ -58,16 +58,42 @@ func generateSnapServiceFile(app *snap.AppInfo) ([]byte, error) {
 	return genServiceFile(app), nil
 }
 
+func stopService(sysd systemd.Systemd, app *snap.AppInfo, inter interacter) error {
+	serviceName := app.ServiceName()
+	tout := serviceStopTimeout(app)
+	if err := sysd.Stop(serviceName, tout); err != nil {
+		if !systemd.IsTimeout(err) {
+			return err
+		}
+		inter.Notify(fmt.Sprintf("%s refused to stop, killing.", serviceName))
+		// ignore errors for kill; nothing we'd do differently at this point
+		sysd.Kill(serviceName, "TERM")
+		time.Sleep(killWait)
+		sysd.Kill(serviceName, "KILL")
+	}
+
+	return nil
+}
+
 // StartSnapServices starts service units for the applications from the snap which are services.
-func StartSnapServices(s *snap.Info, inter interacter) error {
+func StartSnapServices(s *snap.Info, inter interacter) (err error) {
 	sysd := systemd.New(dirs.GlobalRootDir, inter)
+
 	for _, app := range s.Apps {
 		if app.Daemon == "" {
 			continue
 		}
 		if err := sysd.Start(app.ServiceName()); err != nil {
```

Comment:
in the defer `err`'s value is that of the thing being returned. I.e., https://play.golang.org/p/h_MnfP9_Sh